src/v/cluster/scheduling/types.cc

```diff
@@ -208,6 +208,29 @@ bool allocated_partition::has_changes() const {
     return false;
 }
 
+bool allocated_partition::has_node_changes() const {
+    if (!_original) {
+        return false;
+    }
+    if (_replicas.size() != _original->size()) {
+        return true;
+    }
+
+    absl::flat_hash_set<model::node_id> original;
+    original.reserve(_original->size());
+    for (const auto& bs : *_original) {
+        original.insert(bs.node_id);
+    }
+
+    absl::flat_hash_set<model::node_id> current;
+    current.reserve(_replicas.size());
+    for (const auto& bs : _replicas) {
+        current.insert(bs.node_id);
+    }
+
+    return original != current;
```

Comment:
Is it possible that the order of replicas will change but the reallocation will result in the same replica set ? 
If we are certain the order will not be changed we may use `std::mismatch`

Reply:
I think it is possible because new replicas are added to the end. I.e. we try to reallocate replica 0 of [0/0, 1/0, 2/0] and the allocator returns 0/1, so we'll get [1/0, 2/0, 0/1].

So it is better to compare node sets